lib/zinnia.js

```diff
@@ -0,0 +1,128 @@
+import timers from 'node:timers/promises'
+import { execa } from 'execa'
+import { join } from 'node:path'
+import { platform } from 'node:os'
+import * as Sentry from '@sentry/node'
+
+function forwardChunkFromZinnia (chunk, log) {
+  const lines = chunk.trimEnd().split(/\n/g)
+  for (const ln of lines) {
+    log('[ZINNIA] %s', ln)
+  }
+}
+
+async function start ({
+  FIL_WALLET_ADDRESS,
+  STATE_ROOT,
+  CACHE_ROOT,
+  moduleBinaries,
+  metricsStream,
+  activityStream,
+  logStream
+}) {
+  console.log('Starting Zinnia runtime...')
+  logStream.write('Starting Zinnia runtime')
+
+  const zinniadExe = join(moduleBinaries, `zinniad${platform() === 'win32' ? '.exe' : ''}`)
+  const modules = [
+    // all paths are relative to `moduleBinaries`
+    'peer-checker/peer-checker.js'
+  ]
+  const childProcess = execa(zinniadExe, modules, {
+    cwd: moduleBinaries,
+    env: {
+      WALLET_ADDRESS: FIL_WALLET_ADDRESS,
```

Comment:
TODO:
```suggestion
      FIL_WALLET_ADDRESS,
```

After we land & release this Zinnia PR:
- https://github.com/filecoin-station/zinnia/pull/179

Reply:
Approved the PR 👍